src/test/net_tests.cpp

```diff
@@ -135,7 +135,7 @@ BOOST_AUTO_TEST_CASE(cnetaddr_basic)
     CNetAddr addr;
 
     // IPv4, INADDR_ANY
-    BOOST_REQUIRE(LookupHost("0.0.0.0", addr, false));
+    addr = LookupHost("0.0.0.0", false).value();
```

Comment:
No stopper, but I think there should be really a good reason to remove existing unit-test checks. For example, if (for whatever reason) `LookupHost` fails and hence returns std::nullopt, we'd now get a more cryptic error message without seeing the direct line number of the cause:
```
unknown location(0): fatal error: in "net_tests/cnetaddr_basic": std::bad_optional_access: bad_optional_access
test/net_tests.cpp(133): last checkpoint: "cnetaddr_basic" test entry
```
compared to an explicit
```
test/net_tests.cpp(138): fatal error: in "net_tests/cnetaddr_basic": critical check LookupHost("x0.0.0.0", addr, false) has failed
```
(tested by replacing "0.0.0.0" with "x0.0.0.0" both on the PR and master branch). As we pass fixed valid IP addresses here and the probability of this failing is practically zero, I think your approach is still okay though (I'd only see it as problematic if it could lead to undefined behaviour, e.g. by using std::optional's `operator*` instead of `.value()`).

Reply:
I agree that harder to debug test failure messages are to be avoided, but given the nature of the statement, I also don't mind the current approach to avoid having to create a separate var for the `optional` and then the `CNetAddr`, even if it's just the one line. 

It's a shame the `BOOST_` functions don't support pass-through of return values, which would have solved this elegantly. We could use our `Assert` function (requires `#include <util/check.h>`), which yields an equally useful error message:

```c++
    addr = Assert(LookupHost("x0.0.0.0", false)).value();
```

```
test/net_tests.cpp:139 test_method: Assertion `LookupHost("x0.0.0.0", false)' failed.
unknown location:0: fatal error: in "net_tests/cnetaddr_basic": signal: SIGABRT (application abort requested)
test/net_tests.cpp:134: last checkpoint: "cnetaddr_basic" test entry
```

Reply:
Since it's a non-blocker I'm pushing addressing all lastest nits, except this one, to not delay and cause futher rebases conflicts.